### PR TITLE
feat: conversion of ROOT histograms to PlottableHistogram

### PR DESCRIPTION
**Pull request by pieterdavid**
Tested for TH1F by comparing as follows:
```python
import ROOT
h = ROOT.TH1F("h1", "h1", 50, -2.5, 2.5)
h.FillRandom("gaus", 10000)
h.Draw()
from matplotlib import pyplot as plt
fig, ax = plt.subplots()
import mplhep
mplhep.histplot(h)
plt.show()
```
The conversion to numpy exploits the fact that TH1 inherits from TArray (which can directly be converted to a numpy array) for storing the counts, and the sums of squared weights is stored as another TArray (so most could be replaced by equivalent PyROOT pythonizations in the future).
For more-dimensional histograms I put `order="F"` in the reshape such that `.values()[i,j]` equals `.GetBinContent(i+1, j+1)`, but didn't test beyond that. Overflow bins are dropped.

CC @andrzejnovak

### CODE REVIEW

**Comment by henryiii** [comment]
This is awesome, thanks! Bug me if I don't review it by the end of the week.

**Comment by henryiii** [comment]
PS: Sourcery (https://github.com/scikit-hep/uhi/pull/28/commits/c34d8090eaf0790cd4fae4246461060a409d32be) recommends using "guard" style, which has the short-circuit branch come first. It's easier to read with less action at a distance, and avoids excessive indentation.

**Comment by andrzejnovak** [comment]
I was just wondering if this was already in, would be great to have by other :)

**Review by henryiii — APPROVED** [review]
Looks good!  A couple of optional ideas:

* We could either make type aliases to improve readability for the ROOT types (TAxis = Any), or maybe even make them Protocols (TAxis would have GetNbins, etc). But it's really not any different from "Any".
* It's optional and beyond the strict protocol, but maybe we could promote fName to `.name`?

File: src/uhi/numpy_plottable.py [hunk 1]
```diff
@@ -206,5 +323,8 @@ def ensure_plottable_histogram(hist: Any) -> PlottableHistogram:
             # Standard tuple
             return NumPyPlottableHistogram(*(np.asarray(h) for h in hist))
 
+    elif hasattr(hist, "InheritsFrom") and hist.InheritsFrom("TH1"):
```
Comment (henryiii) [inline src/uhi/numpy_plottable.py:326]: TProfile does not insert from TH1, correct?

Comment (pieterdavid) [reply src/uhi/numpy_plottable.py:375]: I didn't consider that, but it does - I added support for it (with the different error types etc. using the arrays directly from numpy isn't as simple in this case, unfortunately, so I just used `GetBinContent` and `GetBinError`)

**Comment by pieterdavid** [comment]
Thanks for the review!

> * We could either make type aliases to improve readability for the ROOT types (TAxis = Any), or maybe even make them Protocols (TAxis would have GetNbins, etc). But it's really not any different from "Any".

I'm not familiar enough with the type annotations yet to tackle that, sorry, but feel free to add it.

> * It's optional and beyond the strict protocol, but maybe we could promote fName to `.name`?

Good idea, done

**Comment by henryiii** [comment]
I've added a ROOT test system (you can even access it with nox, though it will be a bit slow until https://github.com/theacodes/nox/issues/444, and it will be a little harder to maintain unless https://github.com/theacodes/nox/issues/447 gets addressed). Could you add a test converting a ROOT histogram to the mostly-empty test file I've added?

**Comment by henryiii** [comment]
Beautiful, thank you!
